File: old/security/shiro-db/src/main/java/org/tsdes/jee/security/shirodb/UserEntity.java
package org.tsdes.jee.security.shirodb;

import javax.persistence.*;
import javax.validation.constraints.Max;
import javax.validation.constraints.NotNull;
import java.util.List;

/**
 * Created by arcuri82 on 08-Dec-17.
 */
@Entity(name="USERS")
public class UserEntity {

    @Id
    @NotNull
    @Max(255)
    private String username;

    @NotNull
    @Max(255)
    private String password;

    @ElementCollection
    @CollectionTable(name = "USER_ROLES", joinColumns = @JoinColumn(name="username"))
    @Column(name="role_name")
    private List<String>  user_roles;


    public UserEntity() {
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public List<String> getUser_roles() {
        return user_roles;
    }

    public void setUser_roles(List<String> user_roles) {
        this.user_roles = user_roles;
    }
}

File: advanced/rest/exception-handling/src/main/kotlin/org/tsdes/advanced/rest/exceptionhandling/db/UserEntity.kt
package org.tsdes.advanced.rest.exceptionhandling.db

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.validation.constraints.Min
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size


@Entity
class UserEntity(

        @get:NotBlank
        @get:Size(min=0, max=64)
        var name: String,

        @get:NotNull
        @get:Min(0)
        var age: Int,

        @get:Id @get:GeneratedValue
        var id: Long? = null
)


